packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,172 @@
+        relay.instantRelayFeePct === parsedAncillaryData.instantRelayFeePct.toString() &&
+        relay.quoteTimestamp === parsedAncillaryData.quoteTimestamp &&
+        relay.realizedLpFeePct === parsedAncillaryData.realizedLpFeePct.toString() &&
+        relay.slowRelayer.substr(2).toLowerCase() === parsedAncillaryData.slowRelayer &&
+        relay.depositContract.substr(2).toLowerCase() === parsedAncillaryData.depositContract
+    );
+
+    // TODO: Do we need to handle the case where all of these params are matched and matchedRelay.length > 1?
+    if (matchedRelay.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No relay event found for price request time",
+        priceRequestTime: time,
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    const relay = matchedRelay[0];
+
+    // Validate that relay matches a deposit struct exactly.
+    const matchedDeposits = this.deposits.filter(
+      (deposit: Deposit) =>
+        deposit.depositId === relay.depositId &&
+        deposit.depositHash === relay.depositHash &&
+        deposit.timestamp === relay.depositTimestamp &&
+        deposit.sender === relay.sender &&
+        deposit.recipient === relay.recipient &&
+        deposit.l1Token === relay.l1Token &&
+        deposit.amount === relay.amount &&
+        deposit.slowRelayFeePct === relay.slowRelayFeePct &&
+        deposit.instantRelayFeePct === relay.instantRelayFeePct &&
+        deposit.quoteTimestamp === relay.quoteTimestamp
+    );
+    if (matchedDeposits.length > 1) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "TODO: Handle multiple deposits associated with same relay attempt",
+      });
+      throw new Error("TODO: Handle multiple relays for same price request timestamp");
+    } else if (matchedDeposits.length === 0) {
+      this.logger.debug({
+        at: "InsuredBridgePriceFeed",
+        message: "No deposit event matching relay attempt",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // Validate relays proposed realized fee percentage.
+    const expectedRealizedFeePct = this.l1Client.calculateRealizedLpFeesPctForDeposit(/* matchedDeposits[0] */);
+    if (expectedRealizedFeePct !== relay.realizedLpFeePct) {
+      this.logger.error({
+        at: "InsuredBridgePriceFeed",
+        message: "Matched deposit with relay but realized fee % is incorrect",
+      });
+      return this.toBNWei(isRelayValid.No);
+    }
+
+    // TODO: Do we need to check other parameters like slow relayer address and deposit box contract address? These
+    // are the other params included in the ancillary data by the BridgePool contract.
+
+    // Passed all checks, relay is valid!
+    return this.toBNWei(isRelayValid.Yes);
+  }
+
+  public getLastUpdateTime(): number | null {
+    // TODO.
+    return null;
+  }
+
+  public getLookback(): number | null {
+    // TODO.
+    return null;
```

Comment:
Do the clients have any notion of a lookback? It seems like that would be the only limit to the lookback on this price feed.

Reply:
Yeah this is a tricky question, but one I'll save for a follow up PR.